### PR TITLE
Add implicit conversions from java.time classes to plotly LocalDateTime

### PR DESCRIPTION
**Pull request by tmccarthy**
This PR adds implicit conversions from the `java.time` classes to `plotly.element.LocalDateTime`. These conversions are similar to those provided for Joda time in [`plotly.Joda`](https://github.com/alexarchambault/plotly-scala/blob/v0.7.3/joda-time/src/main/scala/plotly/Joda.scala). They are included only in the JVM artefact.

### CODE REVIEW

File: core/jvm/src/main/scala/plotly/JavaTime.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  implicit def fromJavaInstant(javaInstant: Instant): PlotlyLocalDateTime =
+    fromJavaOffsetDateTime(javaInstant.atOffset(ZoneOffset.UTC))
+
+  implicit def fromJavaOffsetDateTime(javaOffsetDateTime: OffsetDateTime): PlotlyLocalDateTime =
+    fromJavaLocalDateTime(javaOffsetDateTime.toLocalDateTime)
+
+  implicit def fromJavaZonedDateTime(javaZonedDateTime: ZonedDateTime): PlotlyLocalDateTime =
+    fromJavaLocalDateTime(javaZonedDateTime.toLocalDateTime)
+
```
Comment (tmccarthy) [inline core/jvm/src/main/scala/plotly/JavaTime.scala:26]: Worth noting that there are some potential gotchas here around converting zoned datetimes caused by dropping the zone component. For example, converting a list of datetimes with differing zones/offsets may lead to unexpected results. I'm happy to add a Scaladoc warning if needed.

Comment (alexarchambault) [reply core/jvm/src/main/scala/plotly/JavaTime.scala:26]: Feel free to mention it in a scaladoc comment if you wish, yes.

**Comment by tmccarthy** [comment]
> Thanks @tmccarthy!
> 
> Maybe the non-ambiguous conversions (from `LocalDateTime`, `Instant`, and `LocalDate`, I guess) could be added directly [in the companion of `plotly.element.LocalDateTime`](https://github.com/alexarchambault/plotly-scala/blob/f0a730dd4218c3f5b85a4eb6b15155e8922629f1/core/shared/src/main/scala/plotly/element/LocalDateTime.scala#L19), so that users don't need the `plotly.JavaTime._` import.

Sounds good. I avoided this because it takes a bit of work to get around the lack of `java.time` classes in ScalaJS, but I will find a better solution.

**Comment by alexarchambault** [comment]
Oh, I had missed that `JavaTime.scala` was under `jvm/`. It can be done by adding a `PlatformLocalDateTimeCompanion` trait for example, which would be empty under `js/`, and have some conversion under `jvm/`. I can do it if you wish.

**Comment by tmccarthy** [comment]
Nah I can do it!

File: core/js/src/main/scala/plotly/element/PlotlyJavaTimeConversions.scala [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package plotly.element
+
+trait PlotlyJavaTimeConversions {
+  // Empty since the java.time classes are not available in ScalaJS
```
Comment (tmccarthy) [inline core/js/src/main/scala/plotly/element/PlotlyJavaTimeConversions.scala:4]: Worth noting that another solution here would be to add a dependency on [`scala-java-time`](https://github.com/cquiroz/scala-java-time) from the core-js project. Let me know if you'd like to do this.

File: core/jvm/src/main/scala/plotly/element/PlotlyJavaTimeConversions.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  implicit def fromJavaLocalDate(javaLocalDate: LocalDate): PlotlyLocalDateTime =
+    fromJavaLocalDateTime(javaLocalDate.atStartOfDay)
+
+  /**
+    * Implicit conversions in this object convert to `plotly.element.LocalDateTime` by simply dropping timezone/offset
+    * information. This can lead to unexpected behaviour, particularly for datasets with varying offsets and timezones.
+    * It will generally be safer to convert your data to `java.time.LocalDateTime` in the appropriate timezone/offset,
+    * and then use the `PlotlyJavaTimeConversions.fromJavaLocalDateTime` implicit conversion.
+    */
+  object UnsafeImplicitConversions {
```
Comment (tmccarthy) [inline core/jvm/src/main/scala/plotly/element/PlotlyJavaTimeConversions.scala:30]: This implicit conversions now need to be explicitly imported using something like:

```scala
import plotly.element.LocalDateTime.UnsafeImplicitConversions._
```

I like this approach, although I am open to feedback on the `UnsafeImplicitConversions` name.

File: core/js/src/main/scala/plotly/element/PlotlyJavaTimeConversions.scala (hunk 2 above)
Comment (alexarchambault) [reply core/js/src/main/scala/plotly/element/PlotlyJavaTimeConversions.scala:4]: I guess it's fine this way, let's not add an extra dependency…

**Comment by alexarchambault** [comment]
That looks great, merging!

**Pull request merged**